internal/config/identity/ldap/ldap.go

```diff
@@ -98,7 +98,7 @@ func (l *Config) GetValidatedDNForUsername(username string) (*xldap.DNSearchResu
 	// under a configured base DN in the LDAP directory.
 	validDN, isUnderBaseDN, err := l.GetValidatedUserDN(conn, username)
 	if err == nil && !isUnderBaseDN {
-		return nil, fmt.Errorf("Unable to find user DN: %w", err)
```

Comment:
why is it not an error? and what happens to the caller if they rely on `err == nil` to make sure xldap.DNSearchResult is non-nil

Reply:
The return of GetValidatedDNForUsername on a non-existent username/DN is `nil, nil`, so caller should not be doing that. We treat users that are not under the search base DN as non-existent, so return the same `nil, nil`.

Reply:
@donatello PTAL not sure what is the context for returning errors here

Reply:
So the question is - is the caller aware of this? That we will be returning nil results with nil error

Please add a comment @taran-p for future users.

Reply:
Added a comment for the change; the overall return behavior is already outlined in the last line of the doc comment